runtime/init-container/src/init.c

```diff
@@ -1927,6 +1957,28 @@ int main(int argc, char **argv) {
     load_module("/9pnet_virtio.ko");
     load_module("/9p.ko");
 
+    if (access("/nvidia.ko", R_OK) == 0) {
+        load_module("/i2c-core.ko");
+        load_module("/drm_panel_orientation_quirks.ko");
+        load_module("/firmware_class.ko");
+        load_module("/drm.ko");
+        load_module("/nvidia.ko");
+        load_module("/nvidia-uvm.ko");
+        load_module("/fbdev.ko");
+        load_module("/fb.ko");
+        load_module("/fb_sys_fops.ko");
+        load_module("/cfbcopyarea.ko");
+        load_module("/cfbfillrect.ko");
+        load_module("/cfbimgblt.ko");
+        load_module("/syscopyarea.ko");
+        load_module("/sysfillrect.ko");
+        load_module("/sysimgblt.ko");
+        load_module("/drm_kms_helper.ko");
+        load_module("/nvidia-modeset.ko");
+        load_module("/nvidia-drm.ko");
+        nvidia_loaded = true;
+    }
```

Comment:
It’s possible to avoid this mess by providing a proper `/sbin/modprobe` for the kernel to use.

Reply:
Proper modprobe could be used directly here, there is no need to call it from the kernel, as modprobe resolves dependencies on its own (based on `modules.dep` or similar file). I'm not sure how much space is a concern, but probably full modprobe + dependencies (glibc etc) is rather undesirable. The one from (statically linked) busybox could be interesting option, but maybe the simplest solution is to move modules list to a text file that is shipped together with the modules (so it can be easily updated if/when module dependencies change)?

Reply:
That’s a good idea, but it should be a separate change.